### PR TITLE
More corner cases

### PR DESCRIPTION
**Pull request by robsdedude**

### CODE REVIEW

File: tests/test_redundant_parentheses.py [hunk 1]
```diff
@@ -637,3 +636,32 @@ def test_double_superfluous_but_helping_parentheses_around_bin_op(
...
+    s = """foo[i:(i + 1)]
+"""
+    assert not plugin(s)
+
+
+# GOOD (redundant in slice, but help readability)
+def test_parens_in_slice_2(plugin):
+    s = """foo[(i - 1):i]
+"""
+    assert not plugin(s)
```
Comment (robsdedude) [inline tests/test_redundant_parentheses.py:652]: Pleas also add those tests

```python
# BAD (redundant in slice and don't help readability)
def test_parens_in_slice_1(plugin):
    s = """foo[i:(-1)]
"""
    assert len(plugin(s)) == 1


# BAD (redundant in slice and don't help readability)
def test_parens_in_slice_2(plugin):
    s = """foo[(0):i]
"""
    assert len(plugin(s)) == 1
```

File: tests/test_redundant_parentheses.py [hunk 2]
```diff
@@ -637,3 +636,32 @@ def test_double_superfluous_but_helping_parentheses_around_bin_op(
         s = f"foo = baz {op2} {parent_expr}"
     s = alteration_func(s)
     assert len(plugin(s)) == 1 + introduced_flakes
+
+
+# GOOD (redundant in slice, but help readability)
+def test_parens_in_slice_1(plugin):
+    s = """foo[i:(i + 1)]
```
Comment (robsdedude) [inline tests/test_redundant_parentheses.py:643]: Of course, this exception should not only work for `+` but for all unary ops, binary ops, boolean operators, etc.
